torch/csrc/jit/autodiff.h

```diff
@@ -86,6 +86,7 @@ Gradient differentiate(std::shared_ptr<Graph>& graph, const std::vector<bool>& r
 // can we take a derivative of this node symbolically?
 bool isDifferentiable(Node * n);
 bool isDifferentiable(Graph & g);
+bool isComparison(Node * n);
```

Comment:
Meh, `isComparison` really isn't something that autodiff should export. Also, it's not like comparison ops are any special for autodiff, the function should really be called sth like `inputRequiresGrad` in here.

Reply:
Agree, did not quite know where to put it, as special handling for comparison ops is also necessary in graph_fuser. But you are right, semantic meaning is different, here it is "output does not need grad", in graph_fuser it is "output does not need to be a float. So have two different lists now, in autodiff and in graph_fuser.